src/view/form-edit-view.js

```diff
@@ -13,139 +15,292 @@ const createFormEditTemplate = (event) => {
+              <div class="event__photos-tape">
+              ${destination.pictures.map((photo) => (`
+                <img class="event__photo" src="${photo.src}" alt="${photo.description}">
+                `)).join('')}
+              </div>
+            </div>
+          </section>
+        </section>
+      </form>
+    </li>
+  `);
 };
 
-export default class FormEditView extends AbstractView {
-  #event = null;
+export default class FormEditView extends AbstractStatefulView {
 
-  constructor(item) {
+  constructor(point) {
     super();
-    this.#event = item;
+    this._state = FormEditView.parseDataToState(point);
+    this.#setInnerHandlers();
   }
 
   get template() {
-    return createFormEditTemplate(this.#event);
+    return createFormEditTemplate(this._state);
   }
 
   setFormSubmitHandler = (callback) => {
     this._callback.formSubmit = callback;
-    this.element.querySelector('form').addEventListener('submit', this.#formSubmitHandler);
-  };
-
-  #formSubmitHandler = (evt) => {
-    evt.preventDefault();
-    this._callback.formSubmit(this.#event);
+    this.element.querySelector('form')
+      .addEventListener('submit', this.#formSubmitHandler);
   };
 
   setDeleteClickHandler = (callback) => {
     this._callback.deleteClick = callback;
-    this.element.querySelector('.event__reset-btn').addEventListener('click', this.#deleteClickHandler);
+    this.element.querySelector('.event__reset-btn')
+      .addEventListener('click', this.#deleteClickHandler);
+  };
+
+  setEditClickHandler = (callback) => {
+    this._callback.editClick = callback;
+    this.element.querySelector('.event__rollup-btn')
+      .addEventListener('click', this.#editClickHandler);
+  };
+
+  reset = (point) => {
+    this.updateElement(
+      FormEditView.parseDataToState(point),
+    );
+  };
+
+  _restoreHandlers = () => {
+    this.#setInnerHandlers();
+    this.setFormSubmitHandler(this._callback.formSubmit);
+    this.setDeleteClickHandler(this._callback.deleteClick);
+    this.setEditClickHandler(this._callback.editClick);
+  };
+
+  #formSubmitHandler = (evt) => {
+    evt.preventDefault();
+    this._callback.formSubmit(FormEditView.parseStateToData(this._state));
   };
 
   #deleteClickHandler = (evt) => {
     evt.preventDefault();
     this._callback.deleteClick();
   };
 
-  setEditClickHandler = (callback) => {
-    this._callback.editClick = callback;
-    this.element.querySelector('.event__rollup-btn').addEventListener('click', this.#editClickHandler);
+  #editClickHandler = (evt) => {
+    evt.preventDefault();
+    this._callback.editClick();
   };
 
-  #editClickHandler = () => {
-    this._callback.editClick();
+  #priceChangeHandler = (evt) => {
+    evt.preventDefault();
+    this._setState({
+      basePrice: evt.target.value
+    });
+  };
+
+  #getChangeDestination = (destinationName) => {
+    const destination = DESTINATIONS.filter((item) => item.name === destinationName);
+
+    if (!destination.length > 0) {
```

Comment:
Лучше не инвертировать условия лишний раз. Это затрудняет восприятие
```suggestion
    if (destination.length === 0) { или if (!destination.length) {

```